### PR TITLE
All dialogs

### PR DESCRIPTION
**Pull request by dr15**
All system dialogs wrapped in their own component. You can see them all in the playground.

### CODE REVIEW

File: src/components/common/WMConfirmationDialog/index.tsx [hunk 1]
```diff
@@ -8,13 +8,19 @@ import { ButtonVariantEnum } from '../WMButton';
 
 import classes from './style.module.scss';
```
Comment (nunibaranes) [inline src/components/common/WMConfirmationDialog/index.tsx:9]: You can't see it here, but please remove the import of CardProps is never read.

File: src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const [selectedEnvironment, setSelectedEnvironment] = useState(environments[0]);
+
+  const handleMenuClick = (selected: IWMDropdownOption) => {
+    setSelectedEnvironment(selected);
+    onChange(selected.text);
+  };
+
+  useEffect(() => {
+    onChange(selectedEnvironment.text);
+  }, []);
```
Comment (nunibaranes) [inline src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx:30]: React Hook useEffect has missing dependencies: 'onChange' and 'selectedEnvironment.text'. Either include them or remove the dependency array. If 'onChange' changes too often, find the parent component that defines it and wrap that definition in useCallback

File: src/components/common/dialogs/LoadingPublishingDialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  open,
+  onClose,
+  environment,
+}: {
+  open: boolean;
+  onClose: () => void;
+  environment: string;
+}): ReactElement {
+  return (
+    <>
```
Comment (nunibaranes) [inline src/components/common/dialogs/LoadingPublishingDialog/index.tsx:19]: I think you don't need the fragment wrapper. I try to remove it and it works :)

File: src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import WMButton from '../../WMButton';
+
+import classes from './style.module.scss';
+
+const environments: IWMDropdownOption[] = [
+  { id: 0, text: 'Production' },
+  { id: 1, text: 'Test' },
+];
+
+export default function EnvironmentDropdown({
```
Comment (nunibaranes) [inline src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx:14]: Maybe we can use the environment-dropdown I create in header-toolbar? not necessary to do it now, it's just an idea for the future.

File: src/components/common/dialogs/PublishToEnvironmentDialog/v.svg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="10" height="9">
+    <path fill="#44BC67" fill-rule="evenodd" d="M8.403 0L3.607 5.685 1.422 3.648 0 5.218 3.788 8.75 10 1.386z"/>
+</svg>
```
Comment (nunibaranes) [inline src/components/common/dialogs/PublishToEnvironmentDialog/v.svg:3]: No newline at the end of the file

File: src/components/common/WMConfirmationDialog/index.tsx (hunk 1 above)
Comment (dr15) [reply src/components/common/WMConfirmationDialog/index.tsx:8]: Fixed

File: src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx (hunk 2 above)
Comment (dr15) [reply src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx]: Fixed by removing it completely and moving the environment up to the parent

File: src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx (hunk 4 above)
Comment (dr15) [reply src/components/common/dialogs/PublishToEnvironmentDialog/EnvironmentDropdown.tsx:9]: I thought about it, but it didn't seem generic enough so I choose to create a separate one

File: src/components/common/dialogs/LoadingPublishingDialog/index.tsx (hunk 3 above)
Comment (dr15) [reply src/components/common/dialogs/LoadingPublishingDialog/index.tsx:19]: Removed!

File: src/components/common/dialogs/PublishToEnvironmentDialog/v.svg (hunk 5 above)
Comment (dr15) [reply src/components/common/dialogs/PublishToEnvironmentDialog/v.svg:3]: Added

**Pull request merged**